dev to main

Th3-Hero opened the pull request:
-Fixed material overlap bug
(Items containing the same material would only show the last calculated due to bad `if` check)
-Logic Rework
-Object Format Rework
-`<title>` fix
-Added Max buttons
-Added Clear Buttons
-Added Tooltip

Th3-Hero committed:
Object and Logic Rework

-Reworked Object and formatting
-Reworked Logic
-Fixed consistency problem

Th3-Hero committed:
Logic Rework, Added Buttons (see desc)

-Fixed material overlap bug
(Items containing the same material would only show the last calculated due to bad `if` check)
-Logic Rework
-Object Format Rework
-`<title>` fix
-Added Max buttons
-Added Clear Buttons
-Added Tooltip

Th3-Hero committed:
Logic Rework, Added Buttons

-Added max/clear buttons to readme

k-seth commented on js/main.js:
```diff
@@ -35,12 +44,12 @@ function toList(collection) {
                 continue;
             }
-            if (!collection.hasOwnProperty(userInput)) {
+            if (!collection.hasOwnProperty(value.material)) {
```
You now use `value.material` 3 times. Might be worth a variable at this point

Th3-Hero replied:
Fixed

k-seth commented on js/main.js:
```diff
@@ -35,12 +44,12 @@ function toList(collection) {
     let collection = {};
     for (let key of Object.keys(crafting)) { // key = key of crafting Object
-        for (let value of crafting[key]) { // value = value associated with key from crafting Object
+        for (let value of crafting[key].materials) { // value = value associated with key from crafting Object
```
The comment is outdated (This is also a place where your variable name could be more descriptive)

Th3-Hero replied:
Fixed naming, comment removed

k-seth commented on css/main.css:
```diff
@@ -169,3 +182,26 @@ label {
     background: #34495e;
 }
+
+.tooltip {
```
As you noted, naming is inconsistent

Th3-Hero replied:
Yes I fixed this as well as the others I was talking about

k-seth commented on js/data.js:
```diff
@@ -1,122 +1,86 @@
-        material: "Nickel"
-    }]
+    heatsink: {
+        max: 150, // max synths per item due to material capacity limit
```
It isn't entirely great form to include comments in objects, but javascript luckily doesn't care (Also, is this comment really needed? I feel like the field name kinda indicates what it is)

Th3-Hero replied:
fixed

k-seth commented on index.html:
```diff
@@ -34,47 +35,79 @@
                     <h1>Anti-Xeno Synthesis Calculator</h1> 
 
-                    <input type="text" id="heatsink" placeholder="0">
+                    <button1 class="clear-input" onclick="clearInput(heatsink)"><span class="material-icons">block</span></button1>
```
you don't define a "clear-input" css class. It would be better if you just used the standard button tag (rather than a whole new one) and then applied the styling changes you want using classes.

Th3-Hero replied:
fixed

Th3-Hero committed:
Various fixes on top of Logic Rework

-Added/Replaced/Renamed some variable names
-Removed outdated comments
-Fixed inconstant naming
-

k-seth commented on css/main.css:
```diff
@@ -140,15 +163,15 @@ label {
 }
 
-.button {
+.clacButton {
```
Sir?

Th3-Hero replied:
fixed

k-seth commented on index.html:
```diff
@@ -3,8 +3,9 @@
-        <title>Anti-Xeno Synthisis Calculator</title>
+        <title>Anti-Xeno Synthesis Calculator</title>
         <link rel="stylesheet" href="css/main.css">
+        <link href="https://fonts.googleapis.com/icon?family=Material+Icons%7CMaterial+Icons" rel="stylesheet">
```
`%7CMaterial+Icons`

Th3-Hero replied:
fixed

Th3-Hero committed:
Final fixes for update

-removed unused stylesheet
-fixed class name typo

Th3-Hero commented:
-Fixed material overlap bug
(Items containing the same material would only show the last calculated due to bad if check)
-Logic Rework
-Object Format Rework
-`<title>` fix
-Added Max buttons
-Added Clear Buttons
-Added Tooltip
-Removed unused stylesheet
-Added/Replaced/Renamed some variable names
-Removed outdated comments
-Fixed inconstant naming
-Updated `README.md`
